app/controllers/api/v1/users_controller.rb

```diff
@@ -33,53 +31,45 @@ def create
       password_confirmation: params[:password_confirmation],
       team_id: params[:team_id],
     )
-    if params[:availability]
-      @user.availability = params[:availability]
+    # creating user availabilities
+    params[:availabilities].each do |avail|
+      @user.availabilities.create!({
+        start: avail[:start],
+        end: avail[:end],
+        somewhat: avail[:somewhat]
+      })
     end
     @team = @user.team
     if @user.save
       bypass_sign_in @user
-      # Change availability from Strings to Integers
-      @user.availability.map! {|arr| arr.map.map(&:to_i) }
-      render json: { status: 'SUCCESS', message: 'User saved and signed in', data: {
-        user: @user,
-        team: @team,
-        captain: @team.captain
-        } }, status: :ok
+
+      data = format_user_data({
+          user: @user.as_json,
+          team: @team.as_json,
+          captain: @team.captain,
+        })
+
+      render json: { status: 'SUCCESS', message: 'User saved and signed in', data: data }, status: :ok
     else
       render json: { status: 'ERROR', message: 'User not saved', data: @user.errors }, status: :unprocessable_entity
     end
   end
 
   # POST /login
   def login
-    validate_login_params
+    helpers.validate_login_params
     @user = User.find_by_email(params[:email])
     if @user&.valid_password?(params[:password])
```

Comment:
I'm not familiar with this notation. What does the & mean in @user&.valid_password?

Reply:
I will add a comment, but essentially it is the same as saying:
```
if @user && @user.valid_password(params[:password])
...
end
```

Reply:
Also, remind me to make a PR to move all controller helper functions to helper classes like line 60 here.